### DISCUSSION TITLE
Reusable components with React

### DISCUSSION THREAD

**Discussion by volnei** [tailwindlabs/headlessui#2705] [General]
Someone have tried to create reusable components wrapping default headlessui keeping its functionality?

**Comment by allysontsoares** [comment]
I'm really trying a shadcn like approach but using headless, but is going so well when comes to integration to react hook form, etc. 
Would be awesome if i have more skills with the tool, but there's almost no content about it on youtube, the docs helps but sometimes is not really helping for some issues. I'm learning, maybe in a few months i will get somewhere close.
